packages/synapse-react-client/src/components/CardMetadataDisplay/CardMetadataDisplay.tsx

```diff
@@ -0,0 +1,159 @@
+import { QueryBundleRequest } from '@sage-bionetworks/synapse-types'
+import {
+  getAdditionalFilters,
+  parseEntityIdFromSqlStatement,
+  SQLOperator,
+} from '../../utils/functions'
+import {
+  BUNDLE_MASK_QUERY_RESULTS,
+  BUNDLE_MASK_QUERY_SELECT_COLUMNS,
+} from '../../utils/SynapseConstants'
+import useGetQueryResultBundle from '../../synapse-queries/entity/useGetQueryResultBundle'
+import { QueryVisualizationWrapperProps } from '../QueryVisualizationWrapper'
+import { SynapseCardLabel } from '../GenericCard'
+import { LabelLinkConfig } from '../CardContainerLogic'
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableRow,
+  Typography,
+} from '@mui/material'
+import { getFieldIndex } from '../../utils/functions/queryUtils'
+
+export type CardMetadataDisplayProps = {
+  sql: string
+  sqlOperator?: SQLOperator
+  searchParams?: Record<string, string>
+  columnLinks?: LabelLinkConfig
+  labels?: string[]
+  columnAliases?: Record<string, string>
+} & Pick<QueryVisualizationWrapperProps, 'additionalFiltersSessionStorageKey'>
+
+const CardMetadataDisplay = (props: CardMetadataDisplayProps) => {
+  const {
+    sql,
+    columnAliases,
+    additionalFiltersSessionStorageKey,
+    sqlOperator,
+    searchParams,
+    columnLinks,
+    labels,
+  } = props
+  const entityId = parseEntityIdFromSqlStatement(sql)
+  const queryFilters = getAdditionalFilters(
+    additionalFiltersSessionStorageKey,
+    searchParams,
+    sqlOperator,
+  )
+
+  const queryBundleRequest: QueryBundleRequest = {
+    partMask: BUNDLE_MASK_QUERY_SELECT_COLUMNS | BUNDLE_MASK_QUERY_RESULTS,
+    concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+    entityId,
+    query: {
+      sql,
+      additionalFilters: queryFilters,
+    },
+  }
+  const { data: queryResultBundle, isLoading } =
+    useGetQueryResultBundle(queryBundleRequest)
+
+  const rowData =
+    queryResultBundle?.queryResult!.queryResults.rows[0].values ?? []
+  const headers = queryResultBundle?.queryResult!.queryResults.headers ?? []
+  const selectColumns = headers.filter(col => labels?.includes(col.name))
+
+  const labelValueArray: [string, string][] = []
+
+  if (labels) {
+    for (let i = 0; i < labels.length; i += 1) {
+      const columnName = labels[i]
+      const columnIndex = getFieldIndex(columnName, queryResultBundle)
+      const value = rowData[columnIndex]
+      // value can be 'null' if column type is of type STRING or '[null]' if column type is of STRING_LIST
+      if (value && value !== '[null]' && value !== 'null') {
+        labelValueArray.push([columnName, value])
+      }
+    }
+  }
+
+  if (isLoading)
+    return (
+      <Typography
+        sx={{
+          fontSize: '14px',
+        }}
+      >
+        Loading...
+      </Typography>
+    )
+
+  return (
+    <div>
```

Comment:
Consider using `StyledTableContainer` to apply styles--you may be able to remove some of your custom styles. You might also have to use html elements instead of MUI components to use the StyledTableContainer styles